packages/block-editor/src/components/link-control/index.js

```diff
@@ -161,7 +163,8 @@ function LinkControl( {
 							key={ `${ suggestion.id }-${ suggestion.type }` }
 							itemProps={ buildSuggestionItemProps( suggestion, index ) }
 							suggestion={ suggestion }
-							onClick={ () => {
+							onClick={ ( event ) => {
+								event.preventDefault();
```

Comment:
There's been a bit of back and forth with this (#18933, #19056), but at least with this edit, it has me wondering: If the intention here is to prevent the form submission caused by `<button type="submit">`, then why should we assign `type="submit"` at all? What are the cases where we _want_ the default behavior to be allowed?

At the very least, if we keep it, we should include some inline comment which explains what we're doing with `preventDefault` and why.

Reply:
Agreed, ideally this onClick should be removed entirely but again because of the weird relation between LinkControl and URLInput we can't. And yes it prevents the submit. See https://github.com/WordPress/gutenberg/pull/19462#discussion_r365880414